File: apps/ios/Pebbles/Features/Record/Steps/RecordValenceStep.swift
import SwiftUI

/// Step 3 — how big and how bright.
///
/// Unlike the other tile steps this one does not advance on pick: the fan is a
/// comparison and the roll is continuous, so `Continue` does the advancing. It
/// arrives parked on neutral-medium so the roll has something to roll.
struct RecordValenceStep: View {
    let model: RecordFlowModel

    var body: some View {
        ValencePickerContent(selected: model.draft.valence) { picked in
            model.select(valence: picked)
        }
        .onAppear { model.seedValenceIfNeeded() }
    }
}

File: apps/android/app/src/main/kotlin/app/pbbls/android/features/path/record/steps/RecordValenceStep.kt
package app.pbbls.android.features.path.record.steps

import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.ui.Modifier
import app.pbbls.android.features.path.models.Valence
import app.pbbls.android.features.path.valence.ValenceFan

/**
 * Step 3 — how big and how bright — the Android analog of iOS
 * `RecordValenceStep`.
 *
 * Renders the fan of nine stones and its two-axis roll, the same content
 * `ValencePickerSheet` shows. Unlike the other tile steps this one does **not**
 * advance on pick: the fan is a comparison and the roll is continuous, so
 * `Continue` does the advancing. It arrives parked on neutral-medium
 * ([onSeed]) so the roll has something to roll.
 */
@Composable
fun RecordValenceStep(
    selected: Valence?,
    onSelect: (Valence) -> Unit,
    onSeed: () -> Unit,
    modifier: Modifier = Modifier,
) {
    LaunchedEffect(Unit) { onSeed() }
    ValenceFan(selected = selected, onSelect = onSelect, modifier = modifier)
}
